storage/storage_image.go

```diff
@@ -682,6 +706,15 @@ func (s *storageImageDestination) Commit(ctx context.Context, unparsedToplevel t
 			lastLayer = layer.ID
 			continue
 		}
+		if ref, ok := s.blobRefs[blob.Digest]; ok {
+			layer, err := s.imageRef.transport.store.PutLayerFromAdditionalLayerStore(id, lastLayer, nil, blob.Digest, ref.String())
```

Comment:
If the `TryReusing…` operation has already looked up the layer, wouldn’t the lookup be able to return some kind of (opaque?) identifier so that here we don’t have to look it up using (digest, ref) again?

Reply:
Fixed to avoid looking up multiple times, by introducing a new object `c/storage.AdditionalLayer` that points to the looked up layer.